src/components/dropdownSearch/DropdownSearch.tsx

```diff
@@ -305,31 +315,56 @@ export class DropdownSearch extends React.Component<IDropdownSearchProps, {}> {
     }
   }
 
+  handleOnClose() {
+    if (this.props.onClose) {
+      this.props.onClose();
+    }
+  }
+
+  private isKeyToHandleOnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
+    return e.keyCode === keyCode.enter
+      || e.keyCode === keyCode.tab
+      || (e.keyCode == keyCode.upArrow && this.props.activeOption === this.props.options[0]);
+  }
+
+  private handleOnOptionClickOnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
+    // prevent onClick because an enter key on an input trigger an click event. OnClick re open the dropdown.
+    // prevent the tab event to select the option and change the focus on next element.
+    // We only want to close the dropdown and get focus on him.
+    // prevent the double focus on input dropdown when key up when the active option
+    // is the first option of the list filter by the filterText
```

Comment:
Since this is a quite long comment, I would use a multi-line comment instead. Also, a few things to increase readability:

* `prevent onClick because pressing the enter key`?
* `triggers a click event`
* `event and OnClick re-opens the dropdown`
* `and get focus on it`
* `on input dropdown on key up`

Reply:
this was a comment made by @pochretien , @pochretien want to reformulate it? I don't want to modify your first intent erroneously

Reply:
its fine for me the rework for the comment 👍